pkg/destroy/aws/aws.go

```diff
@@ -868,9 +869,16 @@ func terminateEC2Instance(ctx context.Context, ec2Client *ec2.EC2, iamClient *ia
 }
 
 func terminateEC2InstanceByInstance(ctx context.Context, ec2Client *ec2.EC2, iamClient *iam.IAM, instance *ec2.Instance, logger logrus.FieldLogger) error {
-	// Skip 'shutting-down' and 'terminated' instances since they take a while to get cleaned up
-	if instance.State == nil || *instance.State.Name == "shutting-down" || *instance.State.Name == "terminated" {
-		return nil
+	// Skip 'terminated' instances since they take a while to get cleaned up
+	if instance.State == nil {
+		if *instance.State.Name == "terminated" {
+			return nil
+		} else if *instance.State.Name == "shutting-down" {
+			if time.Now().Before(timer) {
+				return nil
+			}
+			timer = time.Now().Add(4 * time.Second)
```

Comment:
Since the same timer is used for all instances, the first instance will starve out all of the other instances. In other words, the terminate request will be re-sent for the first instance only until it is terminated.

Reply:
Ah yes of course! Thanks